x-pack/plugins/observability_solution/observability/public/components/alert_search_bar/containers/use_alert_search_bar_state_container.tsx

```diff
@@ -162,43 +161,35 @@ function setupUrlStateSync(
   });
 }
 
-function syncUrlStateWithInitialContainerState(
+function initializeUrlAndStateContainer(
   timefilterService: TimefilterContract,
   stateContainer: AlertSearchBarStateContainer,
   urlStateStorage: IKbnUrlStateStorage,
-  urlStorageKey: string,
-  replace: boolean = true
+  urlStorageKey: string
 ) {
   const urlState = alertSearchBarState.decode(
     urlStateStorage.get<Partial<AlertSearchBarContainerState>>(urlStorageKey)
   );
+  const validUrlState = isRight(urlState) ? pipe(urlState).right : {};
+
+  const { from, to } = timefilterService.getTime();
+  const timeFilterState = timefilterService.isTimeTouched()
+    ? {
+        ...defaultState,
+        rangeFrom: from,
+        rangeTo: to,
+      }
+    : {};
 
-  if (isRight(urlState)) {
-    const newState = {
-      ...defaultState,
-      ...pipe(urlState).right,
-    };
+  const currentState = {
+    ...defaultState,
+    ...timeFilterState,
+    ...validUrlState,
```

Comment:
The order here is the same as we had previously, I just refactored it for readability.